src/Canbus.cpp

```diff
@@ -565,16 +634,43 @@ void sendCanMsg_355()
   }
   else
   {
-    msgData.soc = getBmsChargePercentage(u8_mBmsDatasource); // SOC, uint16 1 %
+    
+    if(WebSettings::getBool(ID_PARAM_INVERTER_OVERWRITE_SOC_EN,0,0,0)==true){
+      uint32_t minVoltage = WebSettings::getInt(ID_PARAM_INVERTER_OVERWRITE_SOC_MIN_VOLTAGE,0,0,0);
+      uint32_t maxVoltage = WebSettings::getInt(ID_PARAM_INVERTER_OVERWRITE_SOC_MAX_VOLTAGE,0,0,0);
+      int32_t soc;
 
-    //Wenn Zellspannung unterschritten wird, dann SoC x an Inverter senden
-    msgData.soc = getNewSocByMinCellVoltage(msgData.soc);
+      soc = ((getBmsAvgVoltage(u8_mBmsDatasource) - minVoltage)*100)/(maxVoltage-minVoltage);
```

Comment:
Ich würde die Berechnung in der `BscSerial::cyclicRun()` machen.
Siehe Kommentar: https://github.com/shining-man/bsc_fw/pull/18/files/7809b9b382e9f61ba3be3e775f88f004600edf24#r1056627240

Reply:
oder gleich nach JkBmsV13_parseData(response);, da sich der Wert erst nach einem parser durchlauf ändern kann.

Reply:
ja klar wenns für alle funktionieren soll dann so wie du gesagt hast